File: app/src/main/java/ru/magflayer/spectrum/presentation/widget/TextSeekBarView.java
package ru.magflayer.spectrum.presentation.widget;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.LayerDrawable;
import android.support.annotation.ColorInt;
import android.support.v7.widget.AppCompatSeekBar;
import android.util.AttributeSet;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.magflayer.spectrum.R;
import ru.magflayer.spectrum.utils.DrawableUtils;

public class TextSeekBarView extends FrameLayout {

    @BindView(R.id.seek_bar)
    protected AppCompatSeekBar seekBar;
    @BindView(R.id.text)
    protected TextView textView;

    public TextSeekBarView(Context context) {
        this(context, null);
    }

    public TextSeekBarView(Context context, AttributeSet attrs) {
        this(context, attrs, -1);
    }

    public TextSeekBarView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    public void setValue(int value) {
        seekBar.setProgress(value);
    }

    public void setMaxValue(int maxValue) {
        seekBar.setMax(maxValue);
    }

    public void setColor(@ColorInt int color) {
        Drawable background = seekBar.getProgressDrawable();
        LayerDrawable layerDrawable = (LayerDrawable) background;
        Drawable drawable = layerDrawable.findDrawableByLayerId(android.R.id.progress);
        DrawableUtils.setColor(drawable, color);
    }

    public void setText(String text) {
        textView.setText(text);
    }

    private void init(Context context) {
        View view = View.inflate(context, R.layout.widget_text_seek_bar, this);
        ButterKnife.bind(this, view);
        seekBar.setOnTouchListener((v, event) -> true);
    }
}

File: app/src/main/java/ru/magflayer/spectrum/presentation/common/android/widget/TextSeekBarView.kt
package ru.magflayer.spectrum.presentation.common.android.widget

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.drawable.LayerDrawable
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.FrameLayout
import androidx.annotation.ColorInt
import ru.magflayer.spectrum.databinding.WidgetTextSeekBarBinding
import ru.magflayer.spectrum.presentation.common.extension.setColor

@SuppressLint("ClickableViewAccessibility")
class TextSeekBarView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = -1
) : FrameLayout(context, attrs, defStyleAttr) {

    private val viewBinding by lazy {
        WidgetTextSeekBarBinding.inflate(
            LayoutInflater.from(context),
            this,
            true
        )
    }

    init {
        viewBinding.seekBar.setOnTouchListener { _, _ -> true }
    }

    fun setValue(value: Int) {
        viewBinding.seekBar.progress = value
    }

    fun setMaxValue(maxValue: Int) {
        viewBinding.seekBar.max = maxValue
    }

    fun setColor(@ColorInt color: Int) {
        val background = viewBinding.seekBar.progressDrawable
        val layerDrawable = background as LayerDrawable
        val drawable = layerDrawable.findDrawableByLayerId(android.R.id.progress)
        drawable.setColor(color)
    }

    fun setText(text: String) {
        viewBinding.text.text = text
    }
}
